packages/react/menu/src/Menu.tsx

```diff
@@ -299,78 +591,229 @@ const ITEM_DEFAULT_TAG = 'div';
 const ITEM_SELECT = 'menu.itemSelect';
 
 type MenuItemOwnProps = Polymorphic.Merge<
-  Omit<Polymorphic.OwnProps<typeof RovingFocusItem>, 'focusable' | 'active'>,
-  {
-    disabled?: boolean;
-    textValue?: string;
-    onSelect?: (event: Event) => void;
-  }
+  Polymorphic.OwnProps<typeof MenuItemImpl>,
+  { onSelect?: (event: Event) => void }
 >;
 
-type MenuItemPrimitive = Polymorphic.ForwardRefComponent<typeof ITEM_DEFAULT_TAG, MenuItemOwnProps>;
+type MenuItemPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof MenuItemImpl>,
+  MenuItemOwnProps
+>;
 
 const MenuItem = React.forwardRef((props, forwardedRef) => {
-  const { as = ITEM_DEFAULT_TAG, disabled = false, textValue, onSelect, ...itemProps } = props;
+  const { disabled = false, onSelect, ...itemProps } = props;
   const ref = React.useRef<HTMLDivElement>(null);
-  const composedRefs = useComposedRefs(forwardedRef, ref);
   const context = useMenuContext(ITEM_NAME);
   const contentContext = useMenuContentContext(ITEM_NAME);
-
-  // get the item's `.textContent` as default strategy for typeahead `textValue`
-  const [textContent, setTextContent] = React.useState('');
-  React.useEffect(() => {
-    const menuItem = ref.current;
-    if (menuItem) {
-      setTextContent((menuItem.textContent ?? '').trim());
-    }
-  }, [itemProps.children]);
-
-  const menuTypeaheadItemProps = useMenuTypeaheadItem({
-    textValue: textValue ?? textContent,
-    disabled,
-  });
+  const composedRefs = useComposedRefs(forwardedRef, ref);
+  const handleSelectProp = useCallbackRef(onSelect);
 
   const handleSelect = () => {
     const menuItem = ref.current;
     if (!disabled && menuItem) {
       const itemSelectEvent = new Event(ITEM_SELECT, { bubbles: true, cancelable: true });
       menuItem.dispatchEvent(itemSelectEvent);
```

Comment:
I think we can simplify this here by doing this:

```suggestion
      const itemSelectEvent = new Event(ITEM_SELECT, { bubbles: true, cancelable: true });
      menuItem.addEventListener(ITEM_SELECT, handleSelectProp, { once: true });
      menuItem.dispatchEvent(itemSelectEvent);
```

Reply:
Yeah def, this is much cleaner.